src/Sylius/Bundle/OrderBundle/Form/DataMapper/OrderItemQuantityDataMapper.php

```diff
@@ -60,8 +60,7 @@ public function mapFormsToData($forms, &$data)
 
         $formsOtherThanQuantity = [];
         foreach ($forms as $key => $form) {
-            if ('quantity' === $form->getName()) {
-                $targetQuantity = $form->getData();
+            if ('quantity' === $form->getName() && $targetQuantity = $form->getData()) {
```

Comment:
But this is modifying `$data` that is then passed to `mapFormsToData`, I'm still investigating but this might bring some ideas.